libs/wingsdk/src/ex/react-website.ts

```diff
@@ -0,0 +1,163 @@
+import { existsSync } from "fs";
+import { isAbsolute, join, resolve } from "path";
+import { Construct } from "constructs";
+import { BaseWebsiteProps, Website } from "../cloud/website";
+import { fqnForType } from "../constants";
+import { App } from "../core";
+import { Resource, Node } from "../std";
+
+const DEFAULT_BUILD_FOLDER = "/build";
+const DEFAULT_START_COMMAND = "npm run start";
+const DEFAULT_BUILD_COMMAND = "npm run build";
+const DEFAULT_PORT = 3001;
+
+/**
+ * Global identifier for `ReactWebsite`.
+ */
+export const REACT_WEBSITE_FQN = fqnForType("ex.ReactWebsite");
+
+/**
+ * Options for `ReactWebsite`.
+ */
+export interface ReactWebsiteProps {
+  /**
+   * The path to the React app root folder- can be absolute or relative to the wing folder
+   */
+  readonly projectPath: string;
+  /**
+   * The path to the React app build folder- relative to the `projectPath`
+   * @default "/build"
+   */
+  readonly buildFolder?: string;
+  /**
+   * A command for starting React app locally
+   * @default "npm run start"
+   */
+  readonly startCommand?: string;
+  /**
+   * A command for building the React app
+   * @default "npm run build"
+   */
+  readonly buildCommand?: string;
+  /**
+   * In sim, if `true` - will use the start command, and if `false` - the build command
+   * @default false
+   */
+  readonly isDevRun?: boolean;
+  /**
+   * Additional properties to run the website host with
+   * @default {}
+   */
+  readonly hostProps?: BaseWebsiteProps;
```

Comment:
Instead of requiring the user pass `Website` props through `hosProps`, can we flatten all of the props? i.e. `ReactWebsiteProps extends BaseWebsiteProps`

Reply:
good idea!

Reply:
unfortunately, it's not easy to take out only the BaseWebsiteProps from all the props and put them into the website (since I can't iterate over interface keys, and I don't want to have to update it manually- it can cause bugs pretty easily)